src/axom/sidre/core/View.cpp

```diff
@@ -114,6 +114,47 @@ View* View::allocate(TypeID type, IndexType num_elems, int allocID)
   return this;
 }
 
+/*
+ *************************************************************************
+ *
+ * Allocate data for view with type and shape.
+ *
+ *************************************************************************
+ */
+View* View::allocate(TypeID type, int ndims, const IndexType* shape, int allocID)
+{
+  allocID = getValidAllocatorID(allocID);
+
```

Comment:
This should have a check for if shape is a nullptr.  Choices for what to do would be to give a SLIC_CHECK_MSG and not allocate, as with the other SLIC_CHECK_MSG conditions here, or to allocate with a default 1D shape.

I notice this issue exists elsewhere in this class -- may be a subsequent git issue should be created to address this more broadly.

Reply:
Done.  Warn but don't allocate if shape is null.  Issue https://github.com/LLNL/axom/issues/1395 submitted for checking other places in the class.